File: sum.rb
# require 'dl'
# require 'dl/import'
require 'fiddle'

# module LibSum
        # extend DL::Importer
        # dlload 'C:/Users/zhaoyang/Desktop/rub/libsum.dll'
        # extern 'double sum(double*, int)'
        # extern 'double split(double)'
# end
libm = Fiddle.dlopen('C:/Users/zhaoyang/Desktop/rub/libsum.dll')
split = Fiddle::Function.new(
    libm['split'],
    [Fiddle::TYPE_DOUBLE],
    Fiddle::TYPE_DOUBLE,
    )

a = [2.0, 3.0, 4.0]
# ss = sum(a.pack("d*"), a.count)
puts split.call(8.8)

File: sum.c
double sum(double *arry, int len)
{
        double ret = 0;
        int i;
        for(i = 0; i < len; i++){
                ret = ret + arry[i];
        }
        return ret;
}
double split(double num)
{
        double ret = 0;
        ret = num / 2;
        return ret;
}
